src/SSVOpenHexagon/Core/HGGraphics.cpp

```diff
@@ -212,11 +242,19 @@ void HexagonGame::draw()
             adjustAlpha(overrideColor, i);
 
             // Draw pivot layers
-            for(std::size_t k = j * numPivotQuads; k < (j + 1) * numPivotQuads;
-                ++k)
+            for(std::size_t k = jAdj * numPivotQuads;
+                k < (jAdj + 1) * numPivotQuads; ++k)
             {
-                pivotQuads3D[k].position += newPos;
-                pivotQuads3D[k].color = overrideColor;
+                if(renderingAbove)
+                {
+                    pivotQuads3DTop[k].position += newPos;
+                    pivotQuads3DTop[k].color = overrideColor;
+                }
+                else
+                {
+                    pivotQuads3D[k].position += newPos;
+                    pivotQuads3D[k].color = overrideColor;
+                }
```

Comment:
Pull the `if` statement outside of the loop, so that we have `if (renderAbove) { for{...} } else { for{...} }`.